sdk/deepbook-v3/src/client.ts

```diff
@@ -43,18 +42,26 @@ export class DeepBookClient {
 		client,
 		address,
 		env,
+		balanceManagers,
 		coins,
 		pools,
 	}: {
 		client: SuiClient;
 		address: string;
 		env: Environment;
+		balanceManagers?: { [key: string]: BalanceManager };
```

Comment:
What's the reason for adding this and removing addBalanceManager? Imo we should abstract away this struct from the user. The purpose of calling functions with keys is that it enables the user not to construct objects outside the client

Reply:
There are almost no scenarios where a user would create a balance manager and then add to it on the fly (since creation only needs to be done once), they are always created before the client is instantiated. For the contracts to function correctly (as seen in the constructor for client.ts), each contract must be initialized with the balance manager configuration. Keeping the option to add a balance manager later would require all contract instances to be reinitialized with the latest configuration, which is not great since the config should be fixed, similar to the pool and coin configs. Items in the configuration shouldn't be updated dynamically after initialization. If updates are necessary, it would be better to manage them within the client instead of the config, however this would make accessing balanceManager Map within contract classes tricky